spec/apis/to-dynamic-path-validator-regex.spec.js

```diff
@@ -0,0 +1,120 @@
+import { toDynamicPathValidatorRegex } from "../../src/applications/apps";
+
+describe("toDynamicPathValidatorRegex", () => {
+  it("Should generate correct regex for '/pathname'", () => {
+    expect(toDynamicPathValidatorRegex("/pathname").test("/pathname")).toBe(
+      true
+    );
+    expect(toDynamicPathValidatorRegex("/pathname").test("/pathname/")).toBe(
+      true
+    );
+    expect(
+      toDynamicPathValidatorRegex("/pathname").test(
+        "/pathname/anything/everything"
+      )
+    ).toBe(true);
+  });
+
+  it("Should generate correct regex for '/#/pathname'", () => {
```

Comment:
Let's add tests for situations where there is both a pathname and a hash. Example `/first/#/second`